libraries/ui/src/utils/addQueryParam.test.ts

```diff
@@ -0,0 +1,38 @@
+import { describe, test, expect } from 'vitest';
+import { addQueryParam } from './addQueryParam';
+
+describe('addQueryParam', () => {
+  test('adds query parameter to URL without existing parameters', () => {
+    const url = 'https://example.com';
+    const result = addQueryParam(url, 'key', 'value');
+    expect(result).toBe('https://example.com/?key=value');
+  });
+
+  test('adds query parameter to URL with existing parameters', () => {
+    const url = 'https://example.com?existing=param';
+    const result = addQueryParam(url, 'key', 'value');
+    expect(result).toBe('https://example.com/?existing=param&key=value');
+  });
+
+  test('adds query parameter to relative path URL', () => {
+    const url = '/login';
+    const result = addQueryParam(url, 'key', 'value');
+    expect(result).toBe('http://localhost:3000/login?key=value');
+  });
+
+  test('handles empty key by returning original URL', () => {
```

Comment:
I think probably we should throw in this case

Reply:
Turns out this is in spec - handled in https://github.com/bluedotimpact/bluedot/pull/741